File: frontend/secret_santa_frontend/src/services/wishlist.service.js
import api from './fetch-api'

export default {
  getWishlist() {
    return api.get('/wishlist')
  },
  addItem(item) {
    return api.post('/wishlist', item)
  },
  updateItem(itemId, item) {
    return api.put(`/wishlist/${itemId}`, item)
  },
  deleteItem(itemId) {
    return api.delete(`/wishlist/${itemId}`)
  }
}

File: backend/src/main/java/ru/secretsanta/service/wishlist/WishlistService.java
package ru.secretsanta.service.wishlist;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import ru.secretsanta.dto.response.WishlistItemResponse;
import ru.secretsanta.entity.wishlist.WishlistItem;

public interface WishlistService {
    WishlistItemResponse addItem(Long userId, WishlistItem item);

    WishlistItemResponse updateItem(Long userId, Long itemId, WishlistItem updatedItem);

    void deleteItem(Long userId, Long itemId);

    Page<WishlistItemResponse> getWishlist(Long userId, Pageable pageable);
}
